helpers/utils.js

```diff
@@ -171,6 +171,15 @@ exports.performAction = async function performAction(request, action) {
             await page.setExtraHTTPHeaders(extraHeaders);
         }
 
-        return await action(page, request);
+        try {
+            return await action(page, request);
+        }
+        catch (err) {
+            throw {
+                contextId: page.browserContext().id,
+                pageId: page.target()._targetId,
+                error: err,
+            }
+        }
```

Comment:
(optional) I'd suggest to [define custom Error subclass](https://github.com/goldbergyoni/nodebestpractices#-22-extend-the-built-in-error-object) or to assign these properties to the original error

Reply:
I decided to assign properties, as this does not duplicate error message